hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,120 @@
+
+- **INSUFFICIENT_GAS**
+  - Not enough gas was supplied to execute the transaction.
+
+- **LOCAL_CALL_MODIFICATION_EXCEPTION**
+  - Local execution (query) is requested for a function that changes state.
+
+- **MAX_CHILD_RECORDS_EXCEEDED**
+  - A contract transaction tried to use more than the allowed number of child records, either through system contract records or internal contract creations.
+
+- **INVALID_CONTRACT_ID**
+  - The contract ID is invalid or does not exist.
+
+- **INVALID_FEE_SUBMITTED**
+  - The fee submitted is invalid.
+
+- **INSUFFICIENT_TX_FEE**
+  - The fee provided in the transaction is insufficient for this type of transaction.
+
+- **INSUFFICIENT_PAYER_BALANCE**
+  - The transaction payer could not afford a custom fee.
+
+- **CONTRACT_EXECUTION_EXCEPTION**
+  - This code is used for any contract execution-related error not handled by specific error codes listed above.
+  - *Example*: An error in Besu due to differences in supported EVM versions. For instance, if Besu supports up to the Shanghai version and the contract is compiled with the Cancun version containing opcode changes, this error would be thrown.
+
+Those are some possible, but very unlikely to happen, top-level status codes. 
+As for MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED and MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED, they indicate too many contracts/entities in the entire system, so the caller can't do anything except wait until the network is fixed.
+MAX_CONTRACT_STORAGE_EXCEEDED applies to an individual contract but still is very hard to reach under normal operation.
+
+- **MAX_CONTRACT_STORAGE_EXCEEDED**
+  - Contract permanent storage exceeded the currently allowable limit.
+
+- **MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED**
+  - All contract storage allocated to the current price regime has been consumed.
+
+- **MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED**
+  - The maximum number of entities allowed in the current price regime have been created.
+
+## Additional statuses, that are not propagated from child transactions, but can be returned as top-level status:
+Those status codes can also be present for `ContractCall`, `ContractCreate` via `EthereumTransaction`.
+
+- **CONTRACT_NEGATIVE_VALUE**
+  - Negative value/initial balance was specified in a smart contract call/create.
+
+- **CONTRACT_DELETED**
+  - Contract is marked as deleted.
+  - Scope: This can happen as a top-level status only for contract call and ethereum contract call.
+
+## Create operation Specific Response Codes
+These responses do include ContractCreate via EthereumTransaction.
+
+- **AUTORENEW_DURATION_NOT_IN_RANGE**
+  - The duration is not a subset of `[MINIMUM_AUTORENEW_DURATION, MAXIMUM_AUTORENEW_DURATION]`.
+
+- **CONTRACT_NEGATIVE_GAS**
+  - Negative gas was offered in a smart contract call.
+
+- **MAX_GAS_LIMIT_EXCEEDED**
+  - Gas exceeded the currently allowable gas limit per transaction.
+
+- **INVALID_MAX_AUTO_ASSOCIATIONS**
```

Comment:
But now there's a well-defined upper limit, right?  Is it worth putting that upper bound here (if that's right)?

Reply:
Currently, INVALID_MAX_AUTO_ASSOCIATIONS also handles the case where we have maxAutomaticTokenAssociations as less than -1.

But there is also REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT which defines the upper limit of 5000 (if you want to go above that, you must set maxAutoAssociations to -1).

I think since we put INVALID_MAX_AUTO_ASSOCIATIONS, it would indeed be useful to also add the other one, for clarity.

I also added NEGATIVE_ALLOWANCE_AMOUNT (which is a little weird because it represents two different contexts; in this case, for Ethereum transactions, it is used for the maxGasAllowance field, but it is also used for the CryptoApproveAllowance operation).